### DISCUSSION TITLE
Bug with api-endpoint GET timesheets

### DISCUSSION THREAD

**Discussion by xstable** [kimai/kimai#4150] [General]
# Version: 1.30.11 stable

I've found a bug in API-Endpoint timesheets. 

If I do request entries for Project with ID 2, I got this   
![image](https://github.com/kimai/kimai/assets/1778068/d529ebeb-625e-4cf5-8a3a-f2b001e60fbe)

(btw. same if I use `?projects=2` as parameter):  
![image](https://github.com/kimai/kimai/assets/1778068/54e283c9-831d-4c06-8672-e7072ff526b0)

If I do request for user-id 3 (which is also part of Project 2) I got this: 

![image](https://github.com/kimai/kimai/assets/1778068/b4abda14-114d-4f27-b55f-5ac36cff43fe)

Backend-Perspective for Timesheets: 

![image](https://github.com/kimai/kimai/assets/1778068/7e49bc57-367d-4a0f-8d66-48647d6bc4ab)

Here the Project-Structure: 

![image](https://github.com/kimai/kimai/assets/1778068/02990e06-7349-48cd-8a96-fbe99c533611)

To be clear. I'd only created "Test 123" by GUI. 
The other entries are added via API: POST timesheets

**Comment by xstable** [comment]
Ahh, I think I see now. 
The documentation missed to noticed, that ONLY the timesheets for the user, who logged in via api, are listed. 

I would expect, if I do `timesheets?project=2`, that I got all timesheets from any user!
But it seems this is no the case. It seems that there is no way to get all timesheets for a project (from all users) with one request... but you have to do a request for each single user that might involved in this projects by `timesheets?user=x` to request their time-sheets. 

Is that right, or am I missed something?

If that is right, we do need a flag for this api like `allUsers` as an dependency to `prjoects`. Means if `allUsers` is true, `projects` must at least contain one value. 
Then it will be possileb to request **all timesheets** for Project 3

**Reply by kevinpapst** [reply]
The API respects the exact same permissions as the UI, would be a huge security problem if not.
You need to use an API user with admin access or promote that API user to be teamlead for each existing team.

**Reply by xstable** [reply]
Hey Kevin, 
thanks for your response. 
I see your point to not show timesheets of all users for the current project by default. Thats a good point!

But to be clear, in my current case, I'm logged in with my Admin-Account to the api, and I've also set my admin-account as Teamleader for all those projects.

**Reply by kevinpapst** [reply]
> User ID to filter timesheets. Needs permission 'view_other_timesheet', pass 'all' to fetch data for all user (default: current user)

<img width="1168" alt="Bildschirmfoto 2023-07-02 um 15 19 31" src="https://github.com/kimai/kimai/assets/533162/b4bb68f2-8fa1-48af-a282-e983bd85ee53">

**Reply by xstable** [reply]
I was able to request the `other users` with my admin-account. But I wasn't able to view all users for this project at once. I need to request them one-by-one via this api-endpoint while changing the user-id.

What I mean would be a request of "all" timesheets for "all users" at this project (if you are admin or teamlead)

**Reply by kevinpapst** [reply]
You want something, Kimai cannot really deliver. 
Timesheets are not bound to Teams and a user could have access to a project through several teams.
If you already have all user IDs, then why don't you use:
<img width="823" alt="Bildschirmfoto 2023-07-03 um 14 21 57" src="https://github.com/kimai/kimai/assets/533162/76a0c8a0-ee71-4b3d-b250-6f22eb38fde1">
